### PR TITLE
Add lock/unlock room functionality

### PR DESCRIPTION
**Pull request by richtrwhereby**
### Description

Add `roomKey`-based authentication controls and allow participants joining with the `host` role to lock/unlock rooms.

**Summary:**

**Related Issue:**

### Testing

1. Check out this branch
2. In the `/.env` file add a valid _host_ `roomKey` for the configured `STORYBOOK_ROOM` in following line:

```
STORYBOOK_ROOM_HOST_ROOMKEY=[host_roomKey_here]
```

3. Run `yarn build && yarn dev`
4. In the Storybook UI (@ http://localhost:6006/) open the story: "Custom UI" > "Room Connection With Host Controls"
5. Check that the `roomKey` is filled with the host room key added in step 3.
6. Clicking on the lock/unlock buttons should now perform the action in the connected room [Screenshot 1].
7. In the Story controls panel, remove the roomKey and reload the page (Cmd + R).
8. The lock/unlock controls shown in the UI should now be in red [Screenshot 2]
9. Clicking on the lock/unlock buttons should not perform the action in the connected room (not connected as a "host" now).
10. Additionally, a warning should be displayed in the console whenever these actions are invoked by a non-"host" user.

#### Screenshot 1: "host" `roomKey` provided

Screenshot 2024-04-05 at 14 37 10

#### Screenshot 2: "host" `roomKey` not provided

Screenshot 2024-04-05 at 14 38 52

### Checklist

-   [x] My code follows the project's coding standards.
-   [x] Prefixed the PR title and commit messages with the service or package name
-   [x] I have written unit tests (if applicable).
-   [ ] I have updated the documentation (if applicable).
-   [x] By submitting this pull request, I confirm that my contribution is made
        under the terms of the MIT license.

### CODE REVIEW

**Review by thyal — COMMENTED** [review]
Nice! Functionality works great, tested with both unauthorized and authorized users. There's a couple of things that should be done before merging though, added some comments. I think we also need to bump media with the type changes.

File: packages/core/src/redux/slices/authorization.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+/**
+ * Selectors
+ */
+
+export const selectAuthorizationRoomKey = (state: RootState) => state.authorization.roomKey;
+
+export const selectAuthorizationRoomLocked = (state: RootState) => state.authorization.roomLocked;
+
+export const selectIsAuthorizedToLockRoom = createSelector(selectLocalParticipantRole, (localParticipantRole) =>
+    AUTHORIZED_ACTION_ROLES.canLockRoom.includes(localParticipantRole as AvailableRoles),
```
[the remark this answers is not in the document]
Comment (richtrwhereby) [reply packages/core/src/redux/slices/authorization.ts]: Fixed in 1ce131c116248a5d4ba2b379ceb43114886c1cb6.

File: packages/core/src/redux/slices/authorization.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
+import { PayloadAction, createSelector, createSlice } from "@reduxjs/toolkit";
+import { RootState } from "../store";
+import { createAppAuthorizedThunk } from "../thunk";
+import { doAppJoin } from "./app";
+import { signalEvents } from "./signalConnection/actions";
+import { selectSignalConnectionRaw } from "./signalConnection";
+import { selectLocalParticipantRole } from "./localParticipant";
+
+type AvailableRoles = "none" | "visitor" | "granted_visitor" | "viewer" | "granted_viewer" | "host";
```
[the remark this answers is not in the document]
Comment (richtrwhereby) [reply packages/core/src/redux/slices/authorization.ts]: Great shout and +1.

All changes added in 1ce131c116248a5d4ba2b379ceb43114886c1cb6.

**Review by thyal — APPROVED** [review]
🍰 Works great and code looks great!

**Pull request merged**